### PR TITLE
Cactus can now grow

### PR DESCRIPTION
**Pull request by Seadragon91**
No wonder I would never see a cactus grow on my skyblock island :)

### CODE REVIEW

**Comment by Seadragon91** [comment]
Ah yes you are right. I also added a change for dropping the cactus, without it you can not create a cactus farm.

**Comment by madmaxoft** [comment]
I hope you tested the changes, I just read the code :)

File: src/Chunk.cpp [hunk 1]
```diff
@@ -1145,20 +1145,33 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
...
+					UnboundedRelGetBlockType(a_RelX + Coord.x, Top + i, a_RelZ + Coord.z, BlockType) &&
 					(
 						cBlockInfo::IsSolid(BlockType) ||
 						(BlockType == E_BLOCK_LAVA) ||
 						(BlockType == E_BLOCK_STATIONARY_LAVA)
 					)
 				)
 				{
+					// Remove cactus
+					GetWorld()->DigBlock(a_RelX + GetPosX() * cChunkDef::Width, Top + i, a_RelZ + GetPosZ() * cChunkDef::Width);
```
Comment (madmaxoft) [inline src/Chunk.cpp]: Doesn't `DigBlock` already create the pickups?

File: src/Chunk.cpp [hunk 2]
```diff
@@ -1145,20 +1145,33 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
...
+					// Drop cactus
+					cBlockHandler* Handler = BlockHandler(E_BLOCK_CACTUS);
+					cChunkInterface ChunkInterface(GetWorld()->GetChunkMap());
+					cBlockInServerPluginInterface PluginInterface(*GetWorld());
+					Handler->DropBlock(
+						ChunkInterface,
+						*GetWorld(),
+						PluginInterface,
+						nullptr,
+						a_RelX + GetPosX() * cChunkDef::Width  + (Coord.x * -1),
```
Comment (madmaxoft) [inline src/Chunk.cpp]: ??? `* -1` ???
Why not just `- Coord.x` ?

Comment (Seadragon91) [reply src/Chunk.cpp]: I want to drop the block on the other side. On the left side is the stone on the right side is air, therefore *-1.

Comment (madmaxoft) [reply src/Chunk.cpp]: That's worth a comment.

Anyway, I just wanted to point out that adding a number multiplied by -1 is the same as subtracting the number.

File: src/Chunk.cpp (hunk 1 above)
Comment (Seadragon91) [reply src/Chunk.cpp]: It doesn't. From the [API](https://api.cuberite.org/cWorld.html#DigBlock_1): `Replaces the specified block with air, without dropping the usual pickups for the block.`

**Comment by Seadragon91** [comment]
The cactus farm I tested works. Now looking why the hopper is not working correctly.

**Comment by madmaxoft** [comment]
If you want to have a look at the hoppers, better do so after #4403 is merged, it changes some stuff in hoppers.

**Pull request merged**